src/main/java/uk/gov/pay/directdebit/mandate/services/gocardless/GoCardlessMandateStateUpdater.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.pay.directdebit.mandate.services.gocardless;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GoCardlessOrganisationId;
+import uk.gov.pay.directdebit.mandate.dao.MandateDao;
+import uk.gov.pay.directdebit.mandate.model.GoCardlessMandateId;
+
+import javax.inject.Inject;
+
+import static java.lang.String.format;
+import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.GOCARDLESS;
+
+public class GoCardlessMandateStateUpdater {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(GoCardlessMandateStateUpdater.class);
+
+    private final MandateDao mandateDao;
+    private final GoCardlessMandateStateCalculator goCardlessMandateStateCalculator;
+
+    @Inject
+    public GoCardlessMandateStateUpdater(MandateDao mandateDao, GoCardlessMandateStateCalculator goCardlessMandateStateCalculator) {
+        this.mandateDao = mandateDao;
+        this.goCardlessMandateStateCalculator = goCardlessMandateStateCalculator;
+    }
+
+    public void updateState(GoCardlessMandateId goCardlessMandateId, GoCardlessOrganisationId goCardlessOrganisationId) {
+        goCardlessMandateStateCalculator.calculate(goCardlessMandateId, goCardlessOrganisationId)
+                .ifPresent(state -> {
+                    int updated = mandateDao.updateStateByPaymentProviderMandateId(GOCARDLESS, goCardlessOrganisationId, goCardlessMandateId, state);
```

Comment:
Could in-line 'update' but perhaps not as expressive e.g...
```
if (mandateDao.updateStateByPaymentProviderMandateId(GOCARDLESS, goCardlessOrganisationId, goCardlessMandateId, state) == 1) {
```

Reply:
Not especially keen on the condition of an `if` statement having a side-effect. I’m not too happy with how this block of code turned out in any case.

Reply:
"Not especially keen on the condition of an if statement having a side-effect" hadn't thought of that. Good point!